src/snowflake/snowpark/functions.py

```diff
@@ -7047,6 +7050,7 @@ def udf(
     external_access_integrations: Optional[List[str]] = None,
     secrets: Optional[Dict[str, str]] = None,
     immutable: bool = False,
+    **kwargs,
```

Comment:
How about raising a ValueError in the code to make sure only `native_app_params` is in the kwargs? This kwargs is tentative so I hope to make sure users don't pass in unexpected arguments. So In the future there won't be a breaking change when we make native_app_params explicit.

Reply:
I prefer just adding `native_app_params` as an optional parameter, why are we using kwargs?

Reply:
The idea was that we'll migrate to an optional parameter once the second half of the implementation is ready. This temporary hiding of the argument is simply to limit accidental discovery in between releases.